File: DeJia/app/src/main/java/com/dejia/anju/api/GetCityApi.java
package com.dejia.anju.api;

import android.content.Context;

import com.dejia.anju.api.base.BaseCallBackApi;
import com.dejia.anju.api.base.BaseCallBackListener;
import com.dejia.anju.net.NetWork;

import java.util.Map;

//获取城市列表
public class GetCityApi implements BaseCallBackApi {
    @Override
    public void getCallBack(Context context, Map<String, Object> maps, final BaseCallBackListener listener) {
        NetWork.getInstance().call("city", "city", maps, context, mData -> listener.onSuccess(mData));
    }
}

File: DeJia/app/src/main/java/com/dejia/anju/api/GetCityApi.kt
package com.dejia.anju.api

import android.content.Context
import com.dejia.anju.api.base.BaseCallBackApi
import com.dejia.anju.api.base.BaseCallBackListener
import com.dejia.anju.net.NetWork
import com.dejia.anju.net.ServerData

//获取城市列表
class GetCityApi : BaseCallBackApi {
    override fun getCallBack(context: Context, maps: MutableMap<String, Any>, listener: BaseCallBackListener<*>) {
        NetWork.instance?.call("city", "city", maps, context) { mData: ServerData? -> listener.onSuccess(mData) }
    }
}